service/worker/scanner/tasklist/scavenger_test.go

```diff
@@ -66,7 +66,9 @@ func (s *ScavengerTestSuite) SetupTest() {
 	s.taskMgr = &mocks.TaskManager{}
 	s.taskListTable = &mockTaskListTable{}
 	s.taskTables = make(map[string]*mockTaskTable)
-	logger := testlogger.New(s.T())
+	zapLogger, err := zap.NewDevelopment()
+	s.Require().NoError(err)
+	logger := loggerimpl.NewLogger(zapLogger)
```

Comment:
[see #5487 unfortunately](https://github.com/uber/cadence/pull/5487#pullrequestreview-1787811461)

Reply:
I would say that introducing a channel that is closed after the log and waiting for the full top should be a better fix.

Reply:
If I land first https://buildkite.com/uberopensource/cadence-server/builds/16817 then please, revert changes.

Reply:
reverting 👍 
possibly unrelated (I don't see any clear signs that it is related) but the first pass had a failure: https://buildkite.com/uberopensource/cadence-server/builds/16871#018c8557-8aae-4026-b49a-34476c703ad7

I've got other changes and retries coming though, could just be other flakiness.  will keep checking.

Reply:
has failed a couple times but did succeed, and overall still seems reasonably stable.  leaving the reverts un-reverted 👍